.github/workflows/cachegrind.yml

```diff
@@ -0,0 +1,47 @@
+name: Valgrind Cachegrind
+
+on:
+  pull_request:
+    branches: [main]
+jobs:
+  valgrind:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup | Rust
+        uses: ATiltedTree/setup-rust@v1
+        with:
+          rust-version: stable
+
+      - name: Install Valgrind
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y valgrind
+
+      - name: Install cg_annotate
+        run: |
+          sudo apt-get install -y kcachegrind
+
+      - name: Run Valgrind
+        run: |
+          cargo b -r -p revm-test --bin snailtracer
+          valgrind --tool=cachegrind target/release/snailtracer
+
+          # Extract relevant information from cachegrind.out file
+          cg_annotate cachegrind.out.* > cachegrind_results.txt
+
+      - name: Valgrind results
+        id: valgrind_results
+        run: |
+          contents=$(printf "%s" "$(head -c 64000 cachegrind_results.txt)")
+          # dump line by line into a file
+          printf "The results of the Valgrind Cachegrind test are (trimmed to only include 64k characters):\n\n\`\`\`\n%s\n\`\`\`" "$contents" > results.md
+
+      - name: Comment on PR
+        env:
+          GH_TOKEN: ${{ github.token }}
```

Comment:
idk but shouldn't we have permission here for token to work? https://docs.github.com/en/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added

Reply:
works perfectly on my fork. i wonder if it would continue to work on another PR?

Reply:
https://github.com/0x0elliot/revm/pull/1 here is my test PR!

Reply:
Yeap we can check after it is merged.

There is  --edit-last flag that could stop the spam. Did you check it out?
https://cli.github.com/manual/gh_pr_comment